### PR TITLE
Remove api prefix from exports

### LINKED ISSUE #295: Export API module at crate level
Currently users of `opentelemetry` must use types from `opentelemetry::api::trace`, we should make the `api` module private and export `trace`, `metrics`, etc, at the crate root to allow the simpler `opentelemetry::trace` to be used in the common case.

### PR DESCRIPTION
**Pull request by jtescher**
Resolves #295

### CODE REVIEW

File: src/api/trace/tracer.rs [hunk 1]
```diff
@@ -83,7 +86,7 @@ use std::time::SystemTime;
 /// greater control over when the span is no longer considered active.
 ///
 /// ```
-/// use opentelemetry::{global, api::trace::{Span, Tracer}};
+/// use opentelemetry::{trace::{Span, Tracer}, global};
```
Comment (TommyCpp) [inline src/api/trace/tracer.rs:89]:
```suggestion
/// use opentelemetry::{global, trace::{Span, Tracer}};
```

File: src/api/trace/tracer.rs [hunk 2]
```diff
@@ -51,7 +53,8 @@ use std::time::SystemTime;
 /// Spans can also use the current thread's [`Context`] to track which span is active:
 ///
 /// ```
-/// use opentelemetry::{global, api::trace::{Tracer, SpanKind}};
+/// use opentelemetry::{global, trace::{Tracer, SpanKind}};
```
Comment (TommyCpp) [inline src/api/trace/tracer.rs:56]:
```suggestion
/// use opentelemetry::{global, trace::{SpanKind, Tracer}};
```

File: src/api/trace/tracer.rs [hunk 3]
```diff
@@ -238,7 +241,7 @@ pub trait Tracer: fmt::Debug + 'static {
     /// # Examples
     ///
     /// ```
-    /// use opentelemetry::{global, api::{trace::{Span, Tracer}, KeyValue}};
+    /// use opentelemetry::{trace::{Span, Tracer}, global, KeyValue};
```
Comment (TommyCpp) [inline src/api/trace/tracer.rs:244]:
```suggestion
    /// use opentelemetry::{global, trace::{Span, Tracer}, KeyValue};
```

File: src/api/trace/tracer.rs [hunk 4]
```diff
@@ -267,7 +270,7 @@ pub trait Tracer: fmt::Debug + 'static {
...
```
Comment (TommyCpp) [inline src/api/trace/tracer.rs:273]:
```suggestion
    /// use opentelemetry::{global, trace::{Span, Tracer}, KeyValue};
```

File: src/api/trace/tracer.rs [hunk 5]
```diff
@@ -302,7 +305,7 @@ pub trait Tracer: fmt::Debug + 'static {
...
```
Comment (TommyCpp) [inline src/api/trace/tracer.rs:308]:
```suggestion
    /// use opentelemetry::{global, trace::{Span, Tracer}, KeyValue};
```

File: src/api/trace/tracer.rs [hunk 6]
```diff
@@ -340,7 +343,7 @@ pub trait Tracer: fmt::Debug + 'static {
     /// # Examples
     ///
     /// ```
-    /// use opentelemetry::{global, api::{trace::{Span, SpanKind, Tracer}, KeyValue}};
+    /// use opentelemetry::{trace::{Span, SpanKind, Tracer}, global, KeyValue};
```
Comment (TommyCpp) [inline src/api/trace/tracer.rs:346]:
```suggestion
    /// use opentelemetry::{global, trace::{Span, SpanKind, Tracer}, KeyValue};
```

File: src/api/trace/tracer.rs [hunk 7]
```diff
@@ -376,7 +379,7 @@ pub trait Tracer: fmt::Debug + 'static {
 ///
 /// ```rust
 /// use opentelemetry::{
-///     api::trace::{TracerProvider, SpanBuilder, SpanKind, Tracer},
+///     trace::{TracerProvider, SpanBuilder, SpanKind, Tracer},
 ///     global,
```
Comment (TommyCpp) [inline src/api/trace/tracer.rs:383]:
```suggestion
///     global,
///     trace::{TracerProvider, SpanBuilder, SpanKind, Tracer},
```

File: src/global/mod.rs [hunk 8]
```diff
@@ -11,12 +11,12 @@
 //! ## Usage
 //!
 //! ```ignore
-//! use opentelemetry::api::trace::{TracerProvider, Tracer};
-//! use opentelemetry::api::metrics::{Meter, MeterProvider};
+//! use opentelemetry::trace::{TracerProvider, Tracer};
```
Comment (TommyCpp) [inline src/global/mod.rs:14]:
```suggestion
//! use opentelemetry::trace::{Tracer, TracerProvider};
```

File: src/lib.rs [hunk 9]
```diff
@@ -13,7 +13,7 @@
 //! ## Getting Started
 //!
 //! ```no_run
-//! use opentelemetry::{api::trace::Tracer, exporter::trace::stdout};
+//! use opentelemetry::{trace::Tracer, exporter::trace::stdout};
```
Comment (TommyCpp) [inline src/lib.rs:16]:
```suggestion
//! use opentelemetry::{exporter::trace::stdout, trace::Tracer};
```

File: README.md [hunk 10]
```diff
@@ -21,7 +21,7 @@ can analyze them using [Prometheus], [Jaeger], and other observability tools.
 ## Getting Started
 
 ```rust
-use opentelemetry::{api::trace::Tracer, exporter::trace::stdout};
+use opentelemetry::{trace::Tracer, exporter::trace::stdout};
```
Comment (TommyCpp) [inline README.md:24]:
```suggestion
use opentelemetry::{exporter::trace::stdout, trace::Tracer};
```

**Review by TommyCpp — APPROVED** [review]
Looks good. A few nits. Rustfmt should really format the example doc code. 😂

**Comment by jtescher** [comment]
Yeah would be nice if rustfmt worked on doc comments for refactors like this 😬

**Pull request merged**